sktime/transformations/series/summarize.py

```diff
@@ -32,14 +32,31 @@ class WindowSummarizer(BaseTransformer):
     lag_feature: dict of str and list, optional (default = dict containing first lag)
         Dictionary specifying as key the type of function to be used and as value
         the argument `window`.
-        For all keys other than `lag`, the argument `window` is a length 2 list
-        containing the integer `lag`, which specifies how far back
-        in the past the window will start, and the integer `window length`,
-        which will give the length of the window across which to apply the function.
-        For ease of notation, for the key "lag", only a single integer
+
+        For all keys other than `lag`, the argument `window` is a length 2 list.
+        The window elements can be either integers indicate the number of
+        observations, or strings specifying a time period (an offset).
+
+        The first element of the `window` is the `lag`, which specifies how far back
+        in the past the window will start. The second element, `window length`,
+        is the length of the window across which to apply the function.
+
+        For ease of notation, for the key "lag", only a single integer or offset
         specifying the `lag` argument will be provided.
 
-        Please see blow a graphical representation of the logic using the following
+        The offsets work only with datasets with time index (either pd.DateimeIndex,
+        pd.RangeIndex, or pd.TimedeltaIndex) are expected to have a format of
+        'NUMBER'+'UNIT', where NUMBER is an integer and UNIT is one of the following:
+        * D - day,
+        * H - hour,
+        * T - minute,
+        * S - second,
+        * L - milisecond,
+        * U - microsecond,
+        * N - nanosecond.
+        For instance, the following offesets would be acceptable: '1D', '2H', '30S'.
+
+        Please see below a graphical representation of the logic using the following
         symbols:
 
         ``z`` = time stamp that the window is summarized *to*.
```

Comment:
random note: should this not be `window_length-1` rather than `1-window_length`?

Reply:
I think it should be `1-window_length`. [This](https://github.com/alan-turing-institute/sktime/blob/4d874c1c20a94d9006604a3916b6b434750b4735/sktime/transformations/series/summarize.py#L46) could cover negative lags (not sure how they are used though).